src/lwc/geBatchGiftEntryHeader/__tests__/geBatchGiftEntryHeader.test.js

```diff
@@ -108,4 +121,56 @@ describe('c-ge-batch-gift-entry-header', () => {
             expect(failedGifts).toBe('5');
         });
     });
+
+    describe('event handling and dispatching', () => {
+        it('should dispatch expected custom event when dry run button is clicked', async () => {
+            const batchHeader = setup();
+            const dispatchEventSpy = jest.spyOn(batchHeader, 'dispatchEvent');
```

Comment:
This is a bit different pattern than what I've gone with for testing customEvents. Since we have a reference to the batchHeader component, we can add event listeners to it directly in the test:

```
const handler = jest.fn();
batchHeader.addEventListener('batchdryrun', handler);
expect(handler).toHaveBeenCalled();
```

Reply:
Updated!